engine/Shopware/Controllers/Frontend/RobotsTxt.php

```diff
@@ -64,7 +64,7 @@ private function getRouterContext(Shop $mainShop): array
 
         foreach ($allShops as $shop) {
             $detachedShop = $shopRepository->getById($shop->getId());
-            if ($detachedShop === null) {
+            if ($detachedShop === null || $detachedShop->getActive() == 0) {
```

Comment:
we should check against false here. this should came back from the shop model. if this does not help, we could still cast it to bool
```suggestion
            if ($detachedShop === null || $detachedShop->getActive() === false) {
```